nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -609,6 +620,90 @@ in
       systemd.targets.zfs.wantedBy = [ "multi-user.target" ];
     })
 
+    (mkIf (cfgZfs.enabled && cfgExpandOnBoot != false) {
+      systemd.services."zpool-expand@" = {
+        description = "Expand ZFS pools";
+        after = [ "zfs.target" ];
+
+        serviceConfig = {
+          Type = "oneshot";
+          RemainAfterExit = true;
+        };
+
+        scriptArgs = "%i";
+        path = [ pkgs.gawk cfgZfs.package ];
+
+        # ZFS has no way of enumerating just devices in a pool in a way
+        # that 'zpool online -e' supports. Thus, we've implemented a
+        # bit of a strange approach of highlighting just devices.
+        # See: https://github.com/openzfs/zfs/issues/12505
+        script = let
+          zpoolScripts = pkgs.writeScriptBin "device-highlighter" ''
+            #!/bin/sh
+
+            # This UUID has been chosen at random and is to provide a
+            # collision-proof, predictable token to search for
+            echo "NIXOS-ZFS-ZPOOL-DEVICE-IDENTIFIER-37108bec-aff6-4b58-9e5e-53c7c9766f05"
```

Comment:
How about defining this magic string in the `let in` and using it in both places, so we can be sure it always matches.